lib/Core/Memory.h

```diff
@@ -377,21 +377,29 @@ class PersistentState : public ObjectState {
      * At any given point, if all cache lines are persisted, we can reset the 
      * set.
      */
-    UpdateList rootCauseLocations;
-    UpdateList pendingRootCauseLocations;
+    UpdateList rootCauseWrites;
+    UpdateList pendingRootCauseWrites;
+    UpdateList rootCauseFlushes;
+    // We could use pending flushes as an indicator for lacking fences.
+    // Would need to clear them.
+    UpdateList pendingRootCauseFlushes;
+    Expr::Width rootCauseWidth;
     // We store all of the unique root cause locations. We can't use the pointer
     // due to copies, but we can make unique IDs
     uint64_t nextLocId = 1; // Start at 1
+    std::string lastCommit;
     std::unordered_map<std::string, uint64_t> allRootLocations;
 
     /// DO NOT USE. Use clone() instead.
     PersistentState(const PersistentState &ps);
 
   public:
 
-    static const uint64_t MaxSize = 4 * (4096);
-    /// Create a new persistent object state from the given non-persistent object state.
-    PersistentState(const ObjectState *os);
+    static uint64_t MaxSize;
+    /// Create a new persistent object state from the given non-persistent
+    /// object state and symbolic bool array of cache lines. Also requires
+    /// a symbolic void* array (int64) for root cause.
+    PersistentState(ExecutionState &state, const ObjectState *os);
```

Comment:
Clarify what the state is needed for.

```suggestion
    /// Create a new persistent object state from the given non-persistent
    /// object state. ExecutionState needed because...
    PersistentState(ExecutionState &state, const ObjectState *os);
```

Reply:
Ack.